### PR TITLE
feat: add a select field on HTTP and gRPC

### PR DESCRIPTION
**Pull request by karatakis**
**Summary:**  
feat: add a select field on HTTP and gRPC

**Issue Reference(s):**  
Fixes https://github.com/tailcallhq/tailcall/issues/2956

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [X] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [X] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [X] I have performed a self-review of my code.
- [X] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/blueprint/operators/http.rs [hunk 1]
```diff
@@ -94,7 +94,20 @@ pub fn compile_http(
...
+            if let Some(select) = &http.select {
+                let dynamic_value = match DynamicValue::try_from(select) {
+                    Ok(dynamic_value) => dynamic_value.prepend("args"),
+                    Err(_) => {
+                        return Valid::fail(format!("syntax error when parsing `{:?}`", select))
+                    }
+                };
+
+                io = io.pipe(IR::Dynamic(dynamic_value));
             }
```
Comment (tusharmath) [inline src/core/blueprint/operators/http.rs:108]: some duplicate code here in http.rs and gRPC. How about we create a new file `select.rs` and move this logic there.

File: src/core/blueprint/operators/http.rs [hunk 2]
```diff
@@ -94,7 +94,20 @@ pub fn compile_http(
                     is_list,
                     dedupe,
                 })
+            };
+
+            if let Some(select) = &http.select {
+                let dynamic_value = match DynamicValue::try_from(select) {
+                    Ok(dynamic_value) => dynamic_value.prepend("args"),
+                    Err(_) => {
```
Comment (tusharmath) [inline src/core/blueprint/operators/http.rs:102]: We should keep the original error also. Try using `Valid::fail_with`.

NOTE: Tests are missing for this branch.

File: src/core/blueprint/dynamic_value.rs [hunk 3]
```diff
@@ -4,14 +4,46 @@ use serde_json::Value;
...
+#[derive(Debug, Clone, PartialEq)]
 pub enum DynamicValue<A> {
     Value(A),
     Mustache(Mustache),
     Object(IndexMap<Name, DynamicValue<A>>),
     Array(Vec<DynamicValue<A>>),
 }
 
+impl<A> DynamicValue<A> {
+    pub fn prepend(self, name: &str) -> Self {
```
Comment (tusharmath) [inline src/core/blueprint/dynamic_value.rs:16]: add doc for this API.

**Review by tusharmath — APPROVED** [review]

**Comment by karatakis** [comment]
docs https://github.com/tailcallhq/tailcallhq.github.io/pull/527

**Pull request merged**